.github/workflows/headercheck.yml

```diff
@@ -0,0 +1,73 @@
+name: Check that headers compile (Libraries and Services)
+
+on:
+  # Automatically run at the end of every day.
+  schedule:
+    - cron: '0 2 * * SAT'
+
+env:
+  # Don't mix these up!
+  # runner.workspace = /home/runner/work/serenity
+  # github.workspace = /home/runner/work/serenity/serenity
+  SERENITY_SOURCE_DIR: ${{ github.workspace }}
+
+jobs:
+  build_header_check:
+    runs-on: ubuntu-22.04
+    if: github.repository == 'SerenityOS/serenity' && github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        # I don't see a point in building HeaderCheck twice, so arbitrarily pick x86_64 as architecture.
+        arch: ['x86_64']
+
+    steps:
+      - uses: actions/checkout@v2
+      # Set default Python to python 3.x, and set Python path such that pip install works properly
+      - uses: actions/setup-python@v2
+
+      - name: "Install Ubuntu dependencies"
+        # These packages are already part of the ubuntu-22.04 image:
+        # cmake libgmp-dev npm shellcheck
+        # Packages below aren't.
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          sudo add-apt-repository 'deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-14 main'
+          sudo apt-get update
+          sudo apt-get install -y clang-format-14 ccache gcc-11 g++-11 libstdc++-11-dev libmpfr-dev libmpc-dev ninja-build unzip
```

Comment:
`clang-format-14` feels like it shouldn't be necessary? Though I don't know if it makes much difference.

Reply:
You're right! I'll also remove ccache, since obviously this build process shouldn't build the same compilation unit twice. I'll keep unzip, because I'm not entirely sure where it's used, and I guess it's a rather unproblematic dependency. The others are most definitely used.